_posts/blog/2016-07-07-reactive_mindset_burgers.md

```diff
@@ -0,0 +1,165 @@
+for(Meat meat: meatList){
+    meat = cook(meat);
+}
+{% endhighlight %}
+
+But what we can do, is to apply a function to every emission of the filtered meat stream, by using the ``map`` operator.
+{% highlight java %}
+Observable.from(meatSource)
+          .filter(meat -> isMeatGoneBad(meat))
+          .map(meat -> cook(meat));
+{% endhighlight %}
+
+## Make a Reactive Burger
+
+Huey, Dewey and Louie create three ``Observable``s: bunObservable, meatObservable and the tomatoObservable. Donald will create the burgerObservable:
+{% highlight java %}
+Observable<Burger> burgerObservable =
+	Observable.zip(bunObservable,
+                  meatObservable,
+                  tomatoObservable,
+                  (bun, meat, tomato) -> makeBurger(bun, meat, tomato));
+{% endhighlight %}
+The ``zip`` operator is the one that will put together one emission of each Observable: a bun, a meat and a tomato slice. In order to do that, it needs to know how to make the burger using the three ingredients. This is defined in the ``makeBurger`` method.
+
+## Mickey Mouse Gets His Burger
+
+If Donald Duck is the one that produces burgers and Mickey Mouse is the one that consumes the burgers, how do the burgers get to Mickey? This is done using the ``subscribe`` method.
+{% highlight java %}
+burgerObservable.subscribe(mickeySubscriber);
+{% endhighlight %}
+Like this, we are saying that every emission of the burgerObservable should be handled by mickeySubscriber.
+
+Mickey Mouse knows how to handle three situations:
+
+* ``onNext(Burger burger)`` - what to do when he gets a burger.
+* ``onError(Throwable error)`` - what to do in case of an error, for example, when the burger was burned.
+* ``onComplete()`` - what to do when the nephews have gone home for the day.
+
+In code, this would look like this:
+{% highlight java %}
+burgerObservable.subscribe(
+		        burger -> eatBurger(burger), //onNext
+		        error -> complain(error), //onError
+		        () -> leaveBurgerJoint()); //onComplete
+{% endhighlight %}
+Every time Mickey Mouse gets a new burger, he will eat it. Whenever something bad happened with the burger, Mickey will complain. When there are no more burgers to be eaten, Mickey Mouse will leave the burger joint.
+
+## Dining Area vs. Kitchen
+Overall, the burger making and consuming would look, in RxJava, like this:
+
+{% highlight java %}
+Observable.zip(bunObservable, meatObservable, tomatoObservable,
+     	          (bun, meat, tomato) -> makeBurger(bun, meat, tomato))
+	  .subscribe(burger -> eatBurger(burger),
+		    error -> complain(error));
+{% endhighlight %}
+
+Mickey Mouse orders the burger in the dining area of the restaurant. By default, in ReactiveX, the event stream would be processed on the same thread. So, this means that the preparation of the burger would also happen in the dining area. But this should actually happen in the kitchen! This also allows Donald Duck to make more burgers while Mickey Mouse eats his burger.
```

Comment:
e more burgers while Mickey Mouse eats his burger - so this happens in parallel.